### PR TITLE
Migrate S4043 to eslint frontend for JS

### LINKED ISSUE #2183: Migrate S4043 to eslint frontend for JS

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #2183 

--
Given the newly raised issues, I would argue that "misleading" mutating calls used as arguments should be considered false positive, and therefore not reported. Since the scope of this PR is only about rule migration, I suggest we create a ticket for it, if my claim makes sense.

### CODE REVIEW

File: its/ruling/src/test/expected/js/backbone/javascript-S4043.json [hunk 1]
```diff
@@ -1,6 +0,0 @@
-{
-'backbone:test/vendor/qunit.js':[
-1809,
```
Comment (andrea-guarino-sonarsource) [inline its/ruling/src/test/expected/js/backbone/javascript-S4043.json:3]: why are those removed?

Comment (yassin-kammoun-sonarsource) [reply its/ruling/src/test/expected/js/backbone/javascript-S4043.json:3]: The implementation of the rule in eslint-bridge defines an exception to ignore mutating calls in return statements, i.e. `return array.sort()`. However, as it is written, the exception is not accurate enough, and thus skips cases like the one you highlighted, which includes such a call with ancestors: call expression < binary expression < return statement.

Comment (andrea-guarino-sonarsource) [reply its/ruling/src/test/expected/js/backbone/javascript-S4043.json]: Thanks for explanation!

**Review by andrea-guarino-sonarsource — APPROVED** [review]
